src/pages/index.js

```diff
@@ -0,0 +1,327 @@
+      previewModalImage.alt = data.name;
+      previewModalCaption.textContent = data.name;
+    });
+
+    cardDeleteBtn.addEventListener('click', () => {
+      openDeleteModal(cardElement);
+    });
+
+    cardImage.src = data.link;
+    cardImage.alt = data.name;
+    cardTitle.textContent = data.name;
+
+    return cardElement;
+  }
+
+
+
+  function handleEditProfileSubmit(evt) {
+    evt.preventDefault();
+    const name = editModalNameInput.value;
+    const about = editModalDescriptionInput.value;
+
+    const saveButton = editProfileForm.querySelector('.modal__submit-btn');
+    saveButton.textContent = 'Saving...';
+    saveButton.disabled = true;
+
+    api.editUserProfile(name, about)
+      .then((userData) => {
+        profileName.textContent = userData.name;
+        profileDescription.textContent = userData.about;
+        closeModal(editProfileModal);
+      })
+      .catch((err) => {
+        console.error('Error updating profile:', err);
+      })
+      .finally(() => {
+        saveButton.textContent = 'Save';
+        saveButton.disabled = false;
+      });
+  }
+
+
+  function handleNewPostSubmit(evt) {
+    evt.preventDefault();
+    const name = newPostCaptionInput.value;
+    const link = newPostLinkInput.value;
+
+    api.createCard(name, link)
+      .then((newCardData) => {
+        const cardElement = getCardElement(newCardData);
+        cardsContainer.prepend(cardElement);
+        closeModal(newPostModal);
+        newPostForm.reset();
+        disableButton(cardSubmitBtn, settings);
+      })
+      .catch((err) => {
+        console.error('Error creating new card:', err);
+      });
+  }
+
+  function openModal(modal) {
+    modal.classList.add('modal_opened');
+
+    const closeModalOverlayClick = (event) => {
+      if (event.target === modal) {
+        closeModal(modal);
+      }
+    };
+    modal.addEventListener('click', closeModalOverlayClick);
+
+    const closeModalOnEsc = (evt) => {
+      if (evt.key === 'Escape') {
+        closeModal(modal);
+      }
+    };
+    document.addEventListener('keydown', closeModalOnEsc);
+
+    modal._closeModalOnClickOutside = closeModalOverlayClick;
+    modal._closeModalOnEsc = closeModalOnEsc;
+  }
+
+  function closeModal(modal) {
+    modal.classList.remove('modal_opened');
+
+    modal.removeEventListener('click', modal._closeModalOnClickOutside);
+    document.removeEventListener('keydown', modal._closeModalOnEsc);
+
+    delete modal._closeModalOnClickOutside;
+    delete modal._closeModalOnEsc;
+  }
+
+  editButton.addEventListener('click', () => {
+    resetValidation(editProfileForm, [editModalNameInput, editModalDescriptionInput]);
+    openModal(editProfileModal);
+    editModalNameInput.value = profileName.textContent;
+    editModalDescriptionInput.value = profileDescription.textContent;
+  });
+
+  newPostButton.addEventListener('click', () => {
+    openModal(newPostModal);
+  });
+
+  newPostCloseButton.addEventListener('click', () => {
```

Comment:
**Could be improved**

You can make a universal handler for any close buttons.

It will look something like this:

```js
// Find all close buttons
const closeButtons = document.querySelectorAll('.modal__close');

closeButtons.forEach((button) => {
  // Find the closest popup only once
  const popup = button.closest('.modal');
  // Set the listener
  button.addEventListener('click', () => closePopup(popup));
});



```

That’s why we make universal css classes for similar elements: here it’s `modal__close` for any close button.

So now you can add 100 modals: their close buttons will be handled automatically.